### DISCUSSION TITLE
Devservices for builds in Containers and Testing

### DISCUSSION THREAD

**Discussion by doogle-oss** [quarkusio/quarkus#23668] [Community]
HI All,
Our CI CD pipelines run within containers and there are no access to docker in containers. Containers are spinned of in kubernetes, copy our code and then tests run. If in need to use test containers which needs a underlying docker how can we go ahead with this. On  container there are no docker daemon. Please advise on this.
#TESTCONTAINERS #DOCKER #TESTING #CI #CD

**Comment by luca-bassoricci** [comment]
I'm facing the same problem with CI/CD in gitlab.
Pipeline compile code with `image: maven:3.6.3-openjdk-11` but I am unable to start a docker daemon to let DevServices start testcontainers correctly.

**Comment by gastaldi** [comment]
In this case I'd recommend to turn off DevServices (with `quarkus.devservices.enabled=false` or `export QUARKUS_DEVSERVICES_ENABLED=false`) and set up the missing values (such as `quarkus.datasource.jdbc.url` or `QUARKUS_DATASOURCE_JDBC_URL` and username / password) before running the tests.

**Reply by doogle-oss** [reply]
We are doing that. Now we use skaffold and kaniko for our builds in kubernetes cluster and the only concern is we use a embedded mongo instead of mongo docker image similarly instead of keycloak we mock the security calls. Overall we do not use the powerful devservices.

**Reply by doogle-oss** [reply]
It's a setback as personally it is for zero setup platform for development and testing in quarkus. I like it a lot but can't use due to ci cd infrastructure

**Reply by raweber42** [reply]
Agreed. This is really annoying, considering that this defeats the purpose of using devservices in the first place.

**Reply by gggoce** [reply]
Any news on this( devservices without dind) or some workaround?
